cmd/cli/serve/util.go

```diff
@@ -28,25 +29,37 @@ import (
 	"github.com/bacalhau-project/bacalhau/pkg/node"
 )
 
-func GetComputeConfig(ctx context.Context, cfg types.NodeConfig, createExecutionStore bool) (node.ComputeConfig, error) {
+func GetComputeConfig(
+	ctx context.Context,
+	cfg types.NodeConfig,
+	r *repo.FsRepo,
+	createExecutionStore bool,
+) (node.ComputeConfig, error) {
 	totalResources, totalErr := cfg.Compute.Capacity.TotalResourceLimits.ToResources()
 	jobResources, jobErr := cfg.Compute.Capacity.JobResourceLimits.ToResources()
 	defaultResources, defaultErr := cfg.Compute.Capacity.DefaultJobResourceLimits.ToResources()
 	if err := errors.Join(totalErr, jobErr, defaultErr); err != nil {
 		return node.ComputeConfig{}, err
 	}
 
-	var err error
 	var executionStore store.ExecutionStore
-
 	if createExecutionStore {
-		executionStore, err = getExecutionStore(ctx, cfg.Compute.ExecutionStore)
+		computeDirPath, err := r.ComputeDir()
+		if err != nil {
+			return node.ComputeConfig{}, err
+		}
+		// TODO(forrest): would it be better to hand the bolt store a path and have it create the database there?
+		executionStore, err = getExecutionStore(ctx, cfg.Compute.ExecutionStore, filepath.Join(computeDirPath, "executions.db"))
```

Comment:
passing a file path should be fine. the file doesn't have to exist as boltdb will create it

Reply:
The directory `executions.db` exists in needs to be created before the path may be handed to boldb, what creates that path?

Reply:
Do you mean `computeDirPath`? It can be created in `node.NewComputeNode(...)` if we start initializing the executionStore inside `node.NewComputeNode()`, or create the parent path here

Reply:
I think I could be in favor of `NewComputeNode` and `NewRequesterNode` being responsible for creating their respective paths. However implementing the code that that gets us there will be nontrivial as it requires re-writing much of our node construction logic. Here's a couple of reasons why:
1. `NewComputeNode` expects a `ComputeConfig`. In order to create a compute config, the ComputePath and ExecutionsPath both must already exist since the `ComputeConfig` contains an `ExecutionStore`, a `LocalPublisher`, and a `PhysicalResourcesProvider` all of which require initialized paths. The `NewComputeNode` method could be modified to create these fields internally, but then we'd lose the ability to mock them out in testing before building a compute node
2. The same is true for `NewRequesterNode`, it accepts a jobStore as input, so it must be initialized before the RequesterNode. Yes we could do it internally, but then we can't configure it differently for testing.
3. Before we initialize either a compute or a requester node, providers for engines, storages, and publishers must exist. Several of these providers require a path on disk to exist. The providers are parameters to `NewComputeNode`, and we frequently override these values with a `NoopProvider` during testing, meaning they can't (yet) be created internally by the compute node based on its config.

My thinking with this PR was: It would be far easier to make a code change where the repo creates these paths on demand, rather than the individual components. In the event we want to make these paths configurable in the future, we can simply add a field to the config and use it in-place of the path provided by the repo.

I am open to suggestion on ways of doing this differently than is done here. But at present, simply adding methods to the config that return paths won't work since they almost always must exist before a compute or requester node is constructed.

Reply:
Move the methods to config, create dirs on demand, and delete repo pkg except for existing migrations